### PR TITLE
[ISSUE #101] Add QueueSelector in producer

### PR DESCRIPTION
**Pull request by xujianhai666**
## What is the purpose of the change

Add QueueSelector in producer

## Brief changelog

add QueueSelector in ProducerOptions
add RoundRobinQueueSelector in default ProducerOptions
replace selector with QueueSelector

## Verifying this change
RoundRobinQueueSelector unit test
add custom manualQueueSelector in example/producer/async

### CODE REVIEW

**Comment by ShannonDing** [comment]
@xujianhai666 Could you please fix the conflicting files?

**Comment by xujianhai666** [comment]
> @xujianhai666 Could you please fix the conflicting files?

@ShannonDing done

File: internal/producer/producer.go [hunk 1]
```diff
@@ -276,9 +274,10 @@ func (p *defaultProducer) buildSendRequest(mq *primitive.MessageQueue,
 	return remote.NewRemotingCommand(kernel.ReqSendMessage, req, msg.Body)
 }
 
-func (p *defaultProducer) selectMessageQueue(topic string) *primitive.MessageQueue {
-	v, exist := p.publishInfo.Load(topic)
+func (p *defaultProducer) selectMessageQueue(msg *primitive.Message) *primitive.MessageQueue {
```
[the remark this answers is not in the document]
Comment (xujianhai666) [reply internal/producer/producer.go]: it seems that "msgs ...*primitive.Message" will not support in this version, we should discuss in next version.

File: primitive/message.go [hunk 2]
```diff
@@ -59,6 +59,10 @@ type Message struct {
 	Properties    map[string]string
 	TransactionId string
 	Batch         bool
+
+	// Queue is the queue that messages will be sent to. the value must be set if want to custom the queue of message,
+	// just ignore if not.
+	Queue         int
```
[the remark this answers is not in the document]
Comment (xujianhai666) [reply primitive/message.go]: done

**Comment by wenfengwang** [comment]
LGTM, @ShannonDing @zongtanghu @jonnxu please merge the PR

**Comment by jonnxu** [comment]
LGTM

**Pull request merged**